source/Calamari.Shared/Integration/Certificates/CryptoKeySecurityAccessRules.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Collections.Generic;
+using Calamari.Integration.Certificates.WindowsNative;
+#if WINDOWS_CERTIFICATE_STORE_SUPPORT
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security.AccessControl;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using static Calamari.Integration.Certificates.WindowsNative.WindowsX509Native;
+#endif
+
+namespace Calamari.Integration.Certificates
+{
+#if WINDOWS_CERTIFICATE_STORE_SUPPORT
+    public static class CryptoKeySecurityAccessRules
```

Comment:
This class is just the extraction of a bunch of CryptoKey logic that was moved out of `WindowsX509CertificateStore` that cannot _currently_ compile in netcore. The functionality is essentially unchanged, with the addition of a compiler directive returning the "failing" implementation to be used by netcore (not actually used-used, but compiled for)